### ISSUE TITLE
Remove ElementAllocatorAlgorithm and KeySpaceTracker after reduce_overhead_data_spec

### ISSUE THREAD

**Issue by Christian-B** [SpiNNakerManchester/PACMAN#465]
is there any reason to have a separate ElementAllocatorAlgorithm when its only use is as a parent to KeySpaceTracker

if you agree it can be merged assign to me.

**Comment by rowleya** [comment]
Agreed.  I will remove from the branch.  There is no PACMAN branch there so that can be removed separately later (so probably leave this open until then).
